### ISSUE TITLE
Document.set_xml_metadata is broken

### ISSUE THREAD

**Issue by MerlijnWajer** [pymupdf/PyMuPDF#3121]
### Description of the bug

Calling `set_xml_metadata` on a PDF Document is broken and results in an error like this:

```
  File "/home/merlijn/archive/archive-pdf-tools/venv/lib/python3.11/site-packages/internetarchivepdf/pdfhacks.py", line 537, in write_metadata
    to_pdf.set_xml_metadata(stream)
  File "/home/merlijn/archive/archive-pdf-tools/venv/lib/python3.11/site-packages/fitz/__init__.py", line 5580, in set_xml_metadata
    xml = mupdf.pdf_add_stream( pdf, res, None, 0)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/merlijn/archive/archive-pdf-tools/venv/lib/python3.11/site-packages/fitz/mupdf.py", line 45040, in pdf_add_stream
    return _mupdf.pdf_add_stream(doc, buf, obj, compressed)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
ValueError: invalid null reference in method 'pdf_add_stream', argument 3 of type 'mupdf::PdfObj const &'
```

Changing line 5580 from ` xml = mupdf.pdf_add_stream( pdf, res, None, 0)` to `xml = mupdf.pdf_add_stream( pdf, res, mupdf.PdfObj(), 0)` seems to "solve" the problem.

### How to reproduce the bug

```
$ python
Python 3.11.7 (main, Jan  5 2024, 14:50:08) [GCC 13.2.1 20230826] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import fitz
>>> doc = fitz.open()
>>> doc.set_xml_metadata('')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/merlijn/archive/archive-pdf-tools/venv/lib/python3.11/site-packages/fitz/__init__.py", line 5580, in set_xml_metadata
    xml = mupdf.pdf_add_stream( pdf, res, None, 0)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/merlijn/archive/archive-pdf-tools/venv/lib/python3.11/site-packages/fitz/mupdf.py", line 45040, in pdf_add_stream
    return _mupdf.pdf_add_stream(doc, buf, obj, compressed)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
ValueError: invalid null reference in method 'pdf_add_stream', argument 3 of type 'mupdf::PdfObj const &'
```

I only used an empty string in the example code, the same error occurs if the string contains valid XML like, something like this:

```
<?xpacket begin="" id="W5M0MpCehiHzreSzNTczkc9d"?>
        <x:xmpmeta xmlns:x="adobe:ns:meta/">
          <rdf:RDF xmlns:rdf="http://www.w3.org/1999/02/22-rdf-syntax-ns#">
            <rdf:Description rdf:about="" xmlns:xmp="http://ns.adobe.com/xap/1.0/">
              <xmp:CreateDate>2023-11-12T22:59:58Z</xmp:CreateDate>
              <xmp:MetadataDate>2023-11-12T22:59:58Z</xmp:MetadataDate>
              <xmp:ModifyDate>2023-11-12T22:59:58Z</xmp:ModifyDate>
              <xmp:CreatorTool>Internet Archive PDF 1.4.21; including mupdf and pymupdf/skimage</xmp:CreatorTool>
            </rdf:Description>
            <rdf:Description rdf:about="" xmlns:pdf="http://ns.adobe.com/pdf/1.3/">
              <pdf:Producer>Internet Archive PDF 1.4.21; including mupdf and pymupdf/skimage</pdf:Producer>
            </rdf:Description>
            <rdf:Description rdf:about="" xmlns:dc="http://purl.org/dc/elements/1.1/">
              <dc:title>
                <rdf:Alt>
                  <rdf:li xml:lang="x-default">Waking up</rdf:li>
                </rdf:Alt>
              </dc:title>
              <dc:creator>
                <rdf:Seq>
                  <rdf:li>Sam Harris</rdf:li>
                </rdf:Seq>
              </dc:creator>
            </rdf:Description>
            <rdf:Description rdf:about="" xmlns:pdfaid="http://www.aiim.org/pdfa/ns/id/">
              <pdfaid:part>3</pdfaid:part>
              <pdfaid:conformance>B</pdfaid:conformance>
            </rdf:Description>
          </rdf:RDF>
        </x:xmpmeta>
        <?xpacket end="r"?>
```

### PyMuPDF version

1.23.20

### Operating system

Linux

### Python version

3.11

**Comment by julian-smith-artifex-com** [comment]
Thanks for the report, your diagnosis is correct. The latest source has a similar fix, which will be included in a new release in the next day or two.

**Comment by julian-smith-artifex-com** [comment]
Fixed in 1.23.21.
